### PR TITLE
Revert unintentional change to BaselineState

### PR DESCRIPTION
**Pull request by lgolding**
The `develop` branch should match TC <span>#</span>30, but we inadvertently introduced a change from  TC <span>#</span>31: replacing `BaselineState.Existing` with `.Unchanged` and `Updated`.

I did not revert the entire change. Some things (like having AppVeyor build the `tc-31` branch instead of the defunct `files-array` branch, and some C# 7 updates to the `PrereleaseCompatibilityTransformer`) were good, and I kept them.

Also:
- Update the version to `2019-01-09` in preparation for merge to `master`.

### CODE REVIEW

**Comment by lgolding** [comment]
@michaelcfanning is added to the review. #Closed

**Comment by lgolding** [comment]
@michaelcfanning I merged this. Feel free to comment _post hoc_. I'm now going to apply the changes from `master` (Bogdan's point fix) to `develop`, in preparation for merging `develop` to `master` and publishing 2019-01-09.

File: src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs [hunk 1]
```diff
@@ -47,28 +47,23 @@ public static class PrereleaseCompatibilityTransformer
...
                 {
-                        modifiedLog |= ApplyChangesFromTC31(sarifLog);
-                        break;
+                    // SARIF TC30. Nothing to do.
+                    break;
                 }
 
                 case "2.0.0-csd.2.beta.2018-10-10":
+                case "2.0.0-csd.2.beta.2018-10-10.1":
+                case "2.0.0-csd.2.beta.2018-10-10.2":
```
Comment (michaelcfanning) [inline src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs]: this is wrong. these aren't valid schema references from the JSON. all these versions should refer to the same JSON schema, the point updates are for the SDK backing them (i.e., the semver associated with the nuget package updates).

**Comment by michaelcfanning** [comment]
@lgolding found a problem but in practice might not be worth fixing (since no one s/be producing SARIF files against the bogus versions referenced).

**Comment by lgolding** [comment]
@michaelcfanning You're right. I will fix it as soon as we decide what branch we're in going forward :-)

**Pull request merged**